src/utils/saml.ts

```diff
@@ -1286,7 +1304,41 @@ export const getPreValidateResponse = (
         )
       ).map(() => _)
     )
-    .bimap(callback, _ => callback(null, true, _.InResponseTo))
+    .bimap(
+      error => {
+        if (eventHandler) {
+          eventHandler({
+            data: {
+              message: error.message
+            },
+            name: "spid.error.generic",
+            type: "ERROR"
+          });
+        }
+        return callback(error);
+      },
+      _ => {
+        // Check if Response Signature is missing
+        const signatureOfResponseCount =
+          _.Response.getElementsByTagNameNS(SAML_NAMESPACE.XMLDSIG, "Signature")
+            .length -
+          _.Assertion.getElementsByTagNameNS(
+            SAML_NAMESPACE.XMLDSIG,
+            "Signature"
+          ).length;
+        if (eventHandler && signatureOfResponseCount === 0) {
+          eventHandler({
+            data: {
+              idpIssuer: _.SAMLRequestCache.idpIssuer,
+              message: "Missing Request signature"
+            },
+            name: "spid.error.signature",
+            type: "INFO"
+          });
+        }
+        return callback(null, true, _.InResponseTo);
```

Comment:
`signatureOfResponseCount` === 0 is considered a successulf case?

Reply:
Yes, the Signature of the Response element is optional ([link](https://docs.italia.it/italia/spid/spid-regole-tecniche/it/stabile/single-sign-on.html#response)), we just log that we have received a SAML response without a signature.

Reply:
why do we check response#signature - assertion#signature ? can you add some comments?

Reply:
Fixed with a comment in 7e7ff8e